### PR TITLE
Coverity: BLAKE oob read

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/blake2/blake2b_mac.c [hunk 1]
```diff
@@ -95,7 +95,8 @@ static int blake2b_mac_ctrl(EVP_MAC_IMPL *macctx, int cmd, va_list args)
             }
             blake2b_param_set_key_length(&macctx->params, (uint8_t)len);
             memcpy(macctx->key, p, len);
-            memset(macctx->key + len, 0, BLAKE2B_KEYBYTES - len);
+            if (len < BLAKE2B_KEYBYTES)
+                memset(macctx->key + len, 0, BLAKE2B_KEYBYTES - len);
```
Comment (kroeckx) [inline crypto/blake2/blake2b_mac.c]: I don't understand this change. The only thing this check seems to do is avoid the call memset(macctx->key + BLAKE2B_KEYBYTES, 0, 0), and I don't see anything wrong with that.

Comment (t-j-h) [reply crypto/blake2/blake2b_mac.c]: It is avoiding a **false positive** report from Coverity prevent. The commit comment should note that this is the context - i.e. that there isn't actually an OOB read.

CID 1442733 (#1 of 1): Out-of-bounds read (OVERRUN)
6. overrun-local: Overrunning array of 64 bytes at byte offset 64 by dereferencing pointer macctx->key + len

Comment (slontis) [reply crypto/blake2/blake2b_mac.c]: Coverity complains about the ptr access. 
The other option is to just ignore it in coverity.

Comment (mattcaswell) [reply crypto/blake2/blake2b_mac.c]: I marked this as a "False positive" in coverity.

Comment (slontis) [reply crypto/blake2/blake2b_mac.c]: ok will close.. thanks

**Pull request closed without merging**